org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationMatching.cs

```diff
@@ -0,0 +1,718 @@
+            var h = 1;
+            foreach ( var header in headers )
+            {
+                worksheet.Cells[1, h].Value = header;
+                h++;
+            }
+            return worksheet;
+        }
+
+        private ExcelWorksheet ColorCell(ExcelWorksheet worksheet, int row, int col)
+        {
+            //Color the Matching Data Green 
+            Color c = System.Drawing.ColorTranslator.FromHtml("#9CD8BC");
+            worksheet.Cells[row, col].Style.Fill.PatternType = ExcelFillStyle.Solid;
+            worksheet.Cells[row, col].Style.Fill.BackgroundColor.SetColor(c);
+            return worksheet;
+        }
+
+        private ExcelWorksheet SaveData(ExcelWorksheet worksheet, int row, Person person, HSContactResult contact)
+        {
+            //Add FirstNames
+            worksheet.Cells[row, 1].Value = contact.properties.firstname;
+            worksheet.Cells[row, 2].Value = person.NickName;
+            if ( person.NickName != person.FirstName )
+            {
+                worksheet.Cells[row, 2].Value += " (" + person.FirstName + ")";
+            }
+            //Color cells if they match
+            if ( CustomEquals(contact.properties.firstname, person.FirstName) || CustomEquals(contact.properties.firstname, person.NickName) )
+            {
+                worksheet = ColorCell(worksheet, row, 1);
+                worksheet = ColorCell(worksheet, row, 2);
+            }
+
+            //Add LastNames
+            worksheet.Cells[row, 3].Value = contact.properties.lastname;
+            worksheet.Cells[row, 4].Value = person.LastName;
+            //Color cells if they match 
+            if ( CustomEquals(contact.properties.lastname, person.LastName) )
+            {
+                worksheet = ColorCell(worksheet, row, 3);
+                worksheet = ColorCell(worksheet, row, 4);
+            }
+
+            //Add Emails
+            worksheet.Cells[row, 5].Value = contact.properties.email;
+            worksheet.Cells[row, 6].Value = person.Email;
+            //Color cells if they match
+            if ( CustomEquals(contact.properties.email, person.Email) )
+            {
+                worksheet = ColorCell(worksheet, row, 5);
+                worksheet = ColorCell(worksheet, row, 6);
+            }
+
+            //Add Phone Numbers
+            var num = person.PhoneNumbers.FirstOrDefault(pn => pn.Number == contact.properties.phone);
+            worksheet.Cells[row, 7].Value = contact.properties.phone;
+            worksheet.Cells[row, 8].Value = num != null ? num.Number : "No Matching Number";
+            //Color cells if they match
+            if ( num != null && CustomEquals(contact.properties.phone, num.Number) )
```

Comment:
The `CustomEquals` method already does a null check, so I believe the `num != null` is redundant and can be removed.